pinot-query-planner/src/main/java/org/apache/pinot/query/QueryEnvironment.java

```diff
@@ -249,16 +253,26 @@ public QueryPlannerResult explainQuery(String sqlQuery, SqlNodeAndOptions sqlNod
           RelNode explainedNode = MultiStageExplainAskingServersUtils.modifyRel(relRoot.rel,
               dispatchableSubPlan.getQueryStages(), nodeTracker, serversExplainer);
 
-          String explainStr = PlannerUtils.explainPlan(explainedNode, format, level);
-
-          return new QueryPlannerResult(null, explainStr, dispatchableSubPlan.getTableNames());
+          return getQueryPlannerResult(plannerContext, dispatchableSubPlan,
+              PlannerUtils.explainPlan(explainedNode, format, level), dispatchableSubPlan.getTableNames());
         }
       }
     } catch (Exception e) {
       throw new RuntimeException("Error explain query plan for: " + sqlQuery, e);
     }
   }
 
+  @NotNull
```

Comment:
(minor) Let's not add `NotNull` annotation, and treat everything as non-null if not annotated